.circleci/config.yml

```diff
@@ -620,17 +620,22 @@ binary_checkout: &binary_checkout
     # Clone the Pytorch branch
     git clone https://github.com/pytorch/pytorch.git "$PYTORCH_ROOT"
     pushd "$PYTORCH_ROOT"
-    if [[ -n "$CIRCLE_PR_NUMBER" ]]; then
-      # "smoke" binary build on PRs
-      git fetch --force origin "pull/${CIRCLE_PR_NUMBER}/head:remotes/origin/pull/${CIRCLE_PR_NUMBER}"
-      git reset --hard "$CIRCLE_SHA1"
-      git checkout -q -B "$CIRCLE_BRANCH"
-      git reset --hard "$CIRCLE_SHA1"
-    elif [[ -n "$CIRCLE_SHA1" ]]; then
-      # "smoke" binary build on master on PR merges
-      git reset --hard "$CIRCLE_SHA1"
-      git checkout -q -B master
-    fi
+    #if [[ -n "$CIRCLE_PR_NUMBER" ]]; then
+    #  # "smoke" binary build on PRs
+    #  git fetch --force origin "pull/${CIRCLE_PR_NUMBER}/head:remotes/origin/pull/${CIRCLE_PR_NUMBER}"
+    #  git reset --hard "$CIRCLE_SHA1"
+    #  git checkout -q -B "$CIRCLE_BRANCH"
+    #  git reset --hard "$CIRCLE_SHA1"
+    #elif [[ -n "$CIRCLE_SHA1" ]]; then
+    #  # "smoke" binary build on master on PR merges
+    #  git reset --hard "$CIRCLE_SHA1"
+    #  git checkout -q -B master
+    #else
+      # nightly binary builds. These run at 05:05 UTC every day. 
+      nightlies_cutoff="$(date +%Y-%m-%d) 05:00"
+      last_commit="$(git log --before '$nightlies_cutoff' -n 1 | perl -lne 'print $1 if /^commit (\w+)/')"
```

Comment:
`git rev-list` instead of `git log` will let you get rid of the perl

Reply:
How would I use a --before with rev-list?

Reply:
`rev-list` works identically to `log` in this regard